api/image.go

```diff
@@ -173,6 +174,14 @@ func (api *API) UpdateImageHandler(w http.ResponseWriter, req *http.Request) {
 	}
 	image.ID = id
 
+	// Acquire lock for image ID, and defer unlocking
+	lockID, err := api.mongoDB.AcquireImageLock(ctx, id)
+	if err != nil {
+		handleError(ctx, w, err, logdata)
+		return
+	}
+	defer api.mongoDB.UnlockImage(lockID)
```

Comment:
An error could be returned from the calls to `api.mongoDB.UnlockImage(lockID)` - should it be handled / logged?

Reply:
maybe it should be logged. The lock will eventually expire, so if unlock fails, future callers can still acquire the lock after it has expired, but we should definitely log it (maybe with WARN level) so that we know there is a problem.